src/views/docs/en/guides/extend/plugins.md

```diff
@@ -154,20 +157,108 @@ rule-two
 
 ... running:
 
-- [`arc create`][create] would create the folders `src/rules/rule-one` and `src/rules/rule-two`, with `index.js` files in each containing the contents of the `body` returned by `pluginFunctions`
+- [`arc create`][create] would create the folders `src/rules/rule-one` and `src/rules/rule-two`, with `index.js` files in each containing the contents of the `body` returned by `functions`
 - [`arc hydrate`][hydrate] would hydrate the above two folders
 - [`arc logs src/rules/rule-one`][logs] would pull in any deployed-to-staging execution logs for the `rule-one` function
 
+### `variables`
+
+> `variables({ arc, cloudformation, stage, inventory })`
+
+The plugin author should implement this method if the plugin would like to provide any manner of data to Lambda functions at runtime. For example, perhaps you would like to expose the physical ID of some AWS resource (i.e. ARN) to your runtime code so that you can interact withit using the AWS SDK.
+
+Architect provides a suite of runtime helpers via the [`@architect/functions`][functions] library. This library leverages functionality provided by [`deploy`][deploy] and [`sandbox`][sandbox] to expose runtime variables enabling [service discovery][discovery] - the automatic configuration, search and discovery of infrastructure and services making up your application. The `variables` plugin method enables plugin authors to hook into the Architect service discovery mechanism.
+
+This method is used by Architect in two situations:
+
+1. When running in a local development context via [`sandbox`][sandbox], `sandbox` will invoke the `variables` method in order to compile all runtime variables required by Architect application plugins and provide them to [`@architect/functions`][functions].
+2. When running [`arc deploy`][deploy], it will invoke the `variables` method in order to compile an [AWS SSM Parameter][ssm] per variable exported by the method. Later, when running in a remotely-deployed runtime context on AWS Lambda, [`@architect/functions`][functions] will query the [AWS SSM Parameter Store][ssm] to retrieve these variables at runtime.
```

Comment:
Not sure we need to discuss SSM and how we're doing this, it's a behind the scenes implementation detail that could change at any time. Not a strong feeling, just a thought.